src/frontend/test_runner/tests/testdata/join.yaml

```diff
@@ -276,3 +278,141 @@
+    LogicalProject { exprs: [c.y, a.x] }
+      LogicalJoin { type: Inner, on: true }
+        LogicalJoin { type: Inner, on: (a.x = b.x) }
+          LogicalScan { table: a, columns: [_row_id, x] }
+          LogicalScan { table: b, columns: [_row_id, x] }
+        LogicalScan { table: c, columns: [_row_id, y] }
+- sql: |
+    /* Ensure that natural joins bind the correct columns */
+    create table a(x int);
+    create table b(x int);
+    select x from a natural join b;
+  logical_plan: |
+    LogicalProject { exprs: [a.x] }
+      LogicalJoin { type: Inner, on: (a.x = b.x) }
+        LogicalScan { table: a, columns: [_row_id, x] }
+        LogicalScan { table: b, columns: [_row_id, x] }
+- sql: |
+    /* Ensure that natural joins bind the correct columns */
+    create table a(x int);
+    create table b(x int);
+    select x from a natural left join b;
+  logical_plan: |
+    LogicalProject { exprs: [a.x] }
+      LogicalJoin { type: LeftOuter, on: (a.x = b.x) }
+        LogicalScan { table: a, columns: [_row_id, x] }
+        LogicalScan { table: b, columns: [_row_id, x] }
+- sql: |
+    /* Ensure that natural joins bind the correct columns */
+    create table a(x int);
+    create table b(x int);
+    select x, a.x, b.x from a natural right join b;
+  logical_plan: |
+    LogicalProject { exprs: [b.x, a.x, b.x] }
+      LogicalJoin { type: RightOuter, on: (a.x = b.x) }
+        LogicalScan { table: a, columns: [_row_id, x] }
+        LogicalScan { table: b, columns: [_row_id, x] }
+- sql: |
+    /* Ensure that natural joins bind the correct columns */
+    create table a(x int);
+    create table b(x int);
+    select x, a.x, b.x from a natural full join b;
+  logical_plan: |
+    LogicalProject { exprs: [Coalesce(a.x, b.x), a.x, b.x] }
+      LogicalJoin { type: FullOuter, on: (a.x = b.x) }
+        LogicalScan { table: a, columns: [_row_id, x] }
+        LogicalScan { table: b, columns: [_row_id, x] }
+- sql: |
+    /* Ensure that nested natural joins bind and disambiguate columns */
+    create table a(x int, y int);
+    create table b(x int, z int);
+    create table c(x int, a int);
+    select x, a.x, b.x, c.x from a natural join b natural join c;
+  logical_plan: |
+    LogicalProject { exprs: [a.x, a.x, b.x, c.x] }
+      LogicalJoin { type: Inner, on: (a.x = c.x) }
+        LogicalJoin { type: Inner, on: (a.x = b.x) }
+          LogicalScan { table: a, columns: [_row_id, x, y] }
+          LogicalScan { table: b, columns: [_row_id, x, z] }
+        LogicalScan { table: c, columns: [_row_id, x, a] }
+- sql: |
+    /* Ensure that nested natural joins bind and disambiguate columns */
+    create table a(x int, y int);
+    create table b(x int, z int);
+    create table c(x int, a int);
+    select x, a.x, b.x, c.x from a natural full join b natural full join c;
+  logical_plan: |
+    LogicalProject { exprs: [Coalesce(a.x, b.x, c.x), a.x, b.x, c.x] }
+      LogicalJoin { type: FullOuter, on: (Coalesce(a.x, b.x) = c.x) }
```

Comment:
Is this really what we want? Or should outer coalesce depend on inner? That way, we'd be able to save on forwarding one column by putting a project in the middle.

But I believe this is functionally correct. 

On another note, I don't know if Coalesce will get pushed down into a project. Else, the middle join will not be an equijoin.

Reply:
We can fix this performance-related issue in future PR.

Reply:
> I don't know if Coalesce will get pushed down into a project. Else, the middle join will not be an equijoin.

Currently, it would not be.

Reply:
Created issue: https://github.com/singularity-data/risingwave/issues/4360

Reply:
#4276